### ISSUE TITLE
Why loss increases and map reduces to zero suddenly at epoch 33?

### ISSUE THREAD

**Issue by 2696120622** [xuannianz/EfficientDet#21]
I train my csv dataset with step1 and step2. At epoch 33 in step2,  loss increases and map reduces to zero suddenly. What should I do for fixing this issue?

Epoch 00030: saving model to checkpoints/2019-12-07/csv_30_0.7954_0.7353.h5
68/68 [==============================] - 129s 2s/step - loss: 0.7954 - regression_loss: 0.6418 - classification_loss: 0.1536 - val_loss: 0.7353 - val_regression_loss: 0.5805 - val_classification_loss: 0.1548
Epoch 31/50
67/68 [============================>.] - ETA: 1s - loss: 0.7850 - regression_loss: 0.6313 - classification_loss: 0.1537Epoch 1/50
Running network: 100% (159 of 159) |######| Elapsed Time: 0:00:21 Time:  0:00:21
Parsing annotations: 100% (159 of 159) |##| Elapsed Time: 0:00:00 Time:  0:00:00
num_fp=14329.0, num_tp=1571.0
157 instances of class livingroom with average precision: 0.9348
90 instances of class bedroom with average precision: 0.9619
297 instances of class room_big with average precision: 0.9398
152 instances of class room_small_long with average precision: 0.7897
849 instances of class door with average precision: 0.9209
36 instances of class other with average precision: 0.1931
mAP: 0.7900

Epoch 00031: saving model to checkpoints/2019-12-07/csv_31_0.7831_0.7654.h5
68/68 [==============================] - 127s 2s/step - loss: 0.7831 - regression_loss: 0.6298 - classification_loss: 0.1533 - val_loss: 0.7654 - val_regression_loss: 0.6131 - val_classification_loss: 0.1523
Epoch 32/50
67/68 [============================>.] - ETA: 1s - loss: 0.7620 - regression_loss: 0.6117 - classification_loss: 0.1503Epoch 1/50
Running network: 100% (159 of 159) |######| Elapsed Time: 0:00:21 Time:  0:00:21
Parsing annotations: 100% (159 of 159) |##| Elapsed Time: 0:00:00 Time:  0:00:00
num_fp=14326.0, num_tp=1574.0
157 instances of class livingroom with average precision: 0.9583
90 instances of class bedroom with average precision: 0.9623
297 instances of class room_big with average precision: 0.9409
152 instances of class room_small_long with average precision: 0.8183
849 instances of class door with average precision: 0.9256
36 instances of class other with average precision: 0.2731
mAP: 0.8131

Epoch 00032: saving model to checkpoints/2019-12-07/csv_32_0.7617_0.7202.h5
68/68 [==============================] - 126s 2s/step - loss: 0.7617 - regression_loss: 0.6105 - classification_loss: 0.1512 - val_loss: 0.7202 - val_regression_loss: 0.5779 - val_classification_loss: 0.1423
Epoch 33/50
67/68 [============================>.] - ETA: 1s - loss: 2.7521 - regression_loss: 1.3013 - classification_loss: 1.4508Epoch 1/50
Running network: 100% (159 of 159) |######| Elapsed Time: 0:00:19 Time:  0:00:19
Parsing annotations: 100% (159 of 159) |##| Elapsed Time: 0:00:00 Time:  0:00:00
num_fp=0, num_tp=0
157 instances of class livingroom with average precision: 0.0000
90 instances of class bedroom with average precision: 0.0000
297 instances of class room_big with average precision: 0.0000
152 instances of class room_small_long with average precision: 0.0000
849 instances of class door with average precision: 0.0000
36 instances of class other with average precision: 0.0000
mAP: 0.0000

Epoch 00033: saving model to checkpoints/2019-12-07/csv_33_2.8166_6.8559.h5
68/68 [==============================] - 124s 2s/step - loss: 2.8166 - regression_loss: 1.3304 - classification_loss: 1.4862 - val_loss: 6.8559 - val_regression_loss: 2.9996 - val_classification_loss: 3.8562
Epoch 34/50
67/68 [============================>.] - ETA: 1s - loss: 2393.7484 - regression_loss: 2389.8921 - classification_loss: 3.8562Epoch 1/50
Running network: 100% (159 of 159) |######| Elapsed Time: 0:00:19 Time:  0:00:19
Parsing annotations: 100% (159 of 159) |##| Elapsed Time: 0:00:00 Time:  0:00:00
num_fp=0, num_tp=0
157 instances of class livingroom with average precision: 0.0000
90 instances of class bedroom with average precision: 0.0000
297 instances of class room_big with average precision: 0.0000
152 instances of class room_small_long with average precision: 0.0000
849 instances of class door with average precision: 0.0000
36 instances of class other with average precision: 0.0000
mAP: 0.0000

Epoch 00034: saving model to checkpoints/2019-12-07/csv_34_2358.6450_6.6235.h5
68/68 [==============================] - 124s 2s/step - loss: 2358.6450 - regression_loss: 2354.7886 - classification_loss: 3.8562 - val_loss: 6.6235 - val_regression_loss: 2.7673 - val_classification_loss: 3.8562
Epoch 35/50
67/68 [============================>.] - ETA: 1s - loss: 7.0168 - regression_loss: 3.1606 - classification_loss: 3.8562Epoch 1/50
Running network: 100% (159 of 159) |######| Elapsed Time: 0:00:19 Time:  0:00:19
Parsing annotations: 100% (159 of 159) |##| Elapsed Time: 0:00:00 Time:  0:00:00
num_fp=0, num_tp=0
157 instances of class livingroom with average precision: 0.0000
90 instances of class bedroom with average precision: 0.0000
297 instances of class room_big with average precision: 0.0000
152 instances of class room_small_long with average precision: 0.0000
849 instances of class door with average precision: 0.0000
36 instances of class other with average precision: 0.0000
mAP: 0.0000

Epoch 00035: saving model to checkpoints/2019-12-07/csv_35_7.0110_6.6163.h5
68/68 [==============================] - 124s 2s/step - loss: 7.0110 - regression_loss: 3.1548 - classification_loss: 3.8562 - val_loss: 6.6163 - val_regression_loss: 2.7601 - val_classification_loss: 3.8562
Epoch 36/50
67/68 [============================>.] - ETA: 1s - loss: 7.7442 - regression_loss: 3.8879 - classification_loss: 3.8562Epoch 1/50
Running network: 100% (159 of 159) |######| Elapsed Time: 0:00:19 Time:  0:00:19
Parsing annotations: 100% (159 of 159) |##| Elapsed Time: 0:00:00 Time:  0:00:00
num_fp=0, num_tp=0
157 instances of class livingroom with average precision: 0.0000
90 instances of class bedroom with average precision: 0.0000
297 instances of class room_big with average precision: 0.0000
152 instances of class room_small_long with average precision: 0.0000
849 instances of class door with average precision: 0.0000
36 instances of class other with average precision: 0.0000
mAP: 0.0000

**Comment by 2696120622** [comment]
There is another issue. I train my csv dataset with step1 and get a model of 36MB. However, I get a model of 82 MB when I train with step2. Why dose the model size increase so many?

**Comment by xuannianz** [comment]
1. The loss diverges during epoch 33.
2. In step 2, there are more trainable parameters, so there are more moving average gradients are saved.

**Comment by 2696120622** [comment]
Thanks.
Then ,what should I do to make the loss decrease continuuously for the rest of training epochs?
